frontend/src/main.tsx

```diff
@@ -2,9 +2,12 @@ import React from "react";
 import ReactDOM from "react-dom/client";
 import App from "./App";
 import "./index.css";
+import { ErrorBoundary } from "react-error-boundary";
 
 ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
+  <ErrorBoundary fallback={<div>Something went wrong</div>}>
```

Comment:
Showing "Something went wrong" is not enough. We should display some error details, as well as suggestion to reload the page or start from the root of the application.

Reply:
I talked to nir regarding what more to do in this pr, I will have an update later this week